consulnameresolver/src/main/java/io/datanerds/grpc/ConsulQueryParameter.java

```diff
@@ -0,0 +1,67 @@
+package io.datanerds.grpc;
+
+import java.util.*;
+
+public class ConsulQueryParameter {
+    public final Optional<String> consulAddress;
+    public final String service;
+    public final Optional<String> datacenter;
+    public final Optional<List<String>> tags;
+
+    private ConsulQueryParameter(Optional<String> consulAddress, String service,
+            Optional<String> datacenter, Optional<List<String>> tags) {
+        this.consulAddress = consulAddress;
+        this.service = service;
+        this.datacenter = datacenter;
+        this.tags = tags;
+    }
+
+    public static class Builder {
+        private String consulAddress;
+        private String service;
+        private String datacenter;
+        private Set<String> tags;
+
+        public Builder(String service) {
+            if (service == null) {
+                throw new RuntimeException("Service cannot be null");
```

Comment:
Is a `RuntimeException` required here? Otherwise I favor a `NullPointerException` and go with

    this.service = Preconditions.checkNotNull(service, "Service cannot be null");